src/components/NewComment.jsx

```diff
@@ -0,0 +1,87 @@
+import { useState } from "react";
+import { addReviewComment } from "../utils/api";
+import Loading from "./Loading";
+
+const NewComment = ({
+  review,
+  username,
+  setCommentSubmitted,
+  setCommCount,
+  comments,
+  setComments,
+}) => {
+  const [inputComment, setInputComment] = useState("");
+  const [newComment, setNewComment] = useState({});
+  const [commentFailed, setCommentFailed] = useState(false);
+  const [noCommentAdded, setNoCommentAdded] = useState(false);
+
+  const { review_id } = review;
+
+  const handleCommentInput = (e) => {
+    setInputComment(e.target.value);
+  };
+
+  const handleSubmitComment = (e) => {
+    setCommentFailed(false);
+    setCommentSubmitted(false);
+    e.preventDefault();
+    const reqObj = { username: username, body: inputComment };
+
```

Comment:
This is where you could've called upon your context (or hardcoded it) instead of passing your username tons of components down